enhancements/insights/up-to-date-gathering.md

```diff
@@ -0,0 +1,250 @@
+for example fast-track, and we would be checking
+```
+quay.io/openshift/origin-insights-operator:4-5-fast-track
+```
+
+Note: I am using origin-insights-operator image, I am not sure if there is a better url to use for OCP image.
+
+For the existence of new manifest. If a new one will be found and it wasn't downloaded yet
+it will proceed with change of image url to the manifest.
+
+For example:
+```
+quay.io/openshift/origin-insights-operator@sha256:606aa2ad67c688934c44d03ec7558cb5ab86d13c30e3363cc14af48b065dac5e
+```
+
+### Update durability
+
+When the *main* container detect new fast-track image in quay and successfully changes
+the *gatherer* container image, it sets it in the support secret in openshift-config, where 
+the rest of the insights operator configuration already lives.
+We will use key "GathererImage"  and set it to particular image url.
+
+For example:
+
+```
+GathererImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:534e829a33e3
+```
+
+This value will be used by *main* update checker. When GathererImage will be latest
+available image in quay it won't update the Gatherer Deployment manifest.
+
+### Returning release image
+
+To revert the original released image, because both containers are running same image,
+we can find the current image of *driver* container with Api alternative to:
+
+For example:
+```
+oc get pod $POD -n openshift-insights  -o=json | jq '.status.containerStatuses[] | { "image": .image, "imageID": .imageID }'
+{
+  "image": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:534e829a33e3f6948720f315ba5894a5edded0d45a8f7981214ebc28a030022c",
+  "imageID": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:534e829a33e3f6948720f315ba5894a5edded0d45a8f7981214ebc28a030022c"
+}
+```
+
+To find exact image, which will be again replaced in Gatherer container.
+
+
+
+### Risks and Mitigations
+
+If download of an image fails because we detect incorrect image, we should be able to 
+return to release image if Pod starts to fail.
+
+
+## Design Details
+
+### <a name="implementation-phases"></a> Implementation phases
+
+1. Implementation of *main* container and its orchestration in driver mode with execution of today’s Release version from Host
+with extracting the Gathering logic to *gatherer* and implementation of its host. At this phase driving will be disabled and 
+there will be no publication to fast-track tag.
+2. We will start publishing changes on fast-track release to Latest version available within the same stream 
+(4.4.z fast-track cluster might get gatherer from 4.4.z2 once available): 4.4.5 gatherer would still need 
+to go though the backporting process to verify it works with that particular stream. Reverting on failure to 
+"release - non fast-track" will be on place.
+3. Unification of fast-track streams. Instead of checking 4.5-fast-track on quay, driver will be looking for unified fast-track.
```

Comment:
This (and 2 above) sound like OLM channels. I'm concerned this enhancement is re-implementing OLM's out of band update model but for a single image within a component that is part of the core release payload. The underlying problem you are describing sounds like you need to install an OLM operator by default as part of a cluster installation.

Reply:
Good points jwforres, we would need to reconsider the approach to be more OLM specific. Thanks for all the good comments !